packages/eslint-plugin/src/rules/enum-utils/shared.ts

```diff
@@ -0,0 +1,46 @@
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+
+import * as util from '../../util';
+
+/*
+ * If passed an enum member, returns the type of the parent. Otherwise,
+ * returns itself.
+ *
+ * For example:
+ * - `Fruit` --> `Fruit`
+ * - `Fruit.Apple` --> `Fruit`
+ */
+export function getBaseEnumType(
```

Comment:
This file is identical in #6091 and has 100% coverage there: https://app.codecov.io/gh/typescript-eslint/typescript-eslint/pull/6091

Reply:
Meh, I don't like the ❌, will change here.